sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -325,15 +326,20 @@ function getKeyValuePairsOrUndefined(
   }
   if (Array.isArray(rawProperties)) {
     for (const rawProperty of rawProperties) {
+      const key = rawProperty.Key;
+      const value = rawProperty.Value["_"];
       const encodedValueType = rawProperty.Value["$"]["i:type"].toString().substring(5);
-      if (encodedValueType === TypeMapForResponseDeserialization.number) {
-        properties[rawProperty.Key] = Number(rawProperty.Value["_"]);
+      if (
+        encodedValueType === TypeMapForResponseDeserialization.int ||
+        encodedValueType === TypeMapForResponseDeserialization.double
+      ) {
+        properties[key] = Number(value);
```

Comment:
Consider a scenario for interop where the value is `4` 
- set using the .NET SDK, gets serialized as "int"
- if the same rule is received using the JS SDK, we lose the information that the value is an "int"
- suppose, the obtained value is serialized again for a new rule using the JS SDK, the value is serialized as a "double"(as opposed to being a "double" originally)
- if someone using the .NET SDK receives the rule, the value is now a "double"


To avoid this (minor) information loss in the interop scenario, We can introduce a new type `{val: {kind: "int", value: number}}` to preserve the int-ness/double-ness properly, and the native "number" type would map to the "double".
But @bterlson and I thought it would be an overkill to introduce a new type for something as trivial as numbers and thought that it would be fine to lose this minor info in the interop scenario. 

/cc Thoughts @chradek @JoshLove-msft @hemanttanwar @KieranBrantnerMagee

Reply:
In .NET, the AMQP library has an [implementation ](https://github.com/Azure/azure-amqp/tree/master/src/Encoding) for each of the [supported numeric types](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#section-primitive-type-definitions), so there should not be a loss of type information.

Reply:
This isn't AMQP, this is referring to the ATOM API.

Reply:
Ah..
Here is the [relevant ](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/Rules/XmlObjectConvertor.cs) .NET code.
We include the type information in the XML - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/tests/SessionRecords/ServiceBusManagementClientLiveTests/BasicRuleCrudOperationsAsync.json#L148

Reply:
We also include the type info. 
However, due to the limitation on the types in JS(TS), we can't differentiate "int" vs "double", and that leads to considering both "int" and "double" as number during the deserialization, and serializing numbers as "double". 

If a number serialized as "int" using the .NET SDK, is received and re-used to serialize again through the JS SDK, it gets serialized as a "double" though it was an "int" originally.

Reply:
Ah thanks - yes I was thinking you were asking if this was an issue in other libraries rather than asking for our opinions. I agree with your decision to not introduce a new type - I wouldn't think that anyone would complain that we are representing an int as a double - they are not losing any precision.